mobile/mobile-core/devices-service.ts

```diff
@@ -392,19 +392,34 @@ export class DevicesService extends EventEmitter implements Mobile.IDevicesServi
 			}
 			let deviceInstances = this.getDeviceInstances();
 
-			//if no --device is passed and no devices are found, the default emulator is started
 			if (!data.deviceId && _.isEmpty(deviceInstances)) {
 				if (!this.$hostInfo.isDarwin && this.$mobileHelper.isiOSPlatform(data.platform)) {
 					this.$errors.failWithoutHelp(constants.ERROR_NO_DEVICES_CANT_USE_IOS_SIMULATOR);
 				}
+			}
+
+			try {
+				await this._startEmulatorIfNecessary(data);
+			} catch (err) {
+				const errorMessage = this.getEmulatorError(err, data.platform);
 
+				this.$errors.failWithoutHelp(errorMessage);
+			}
+		}
+	}
+
+	private async _startEmulatorIfNecessary(data?: Mobile.IDevicesServicesInitializationOptions): Promise<void> {
+			let deviceInstances = this.getDeviceInstances();
```

Comment:
I'd prefer that instead of passing it as an argument, but if you insist I can change it...